pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -1655,6 +1656,34 @@ void expireMessagesForAllSubscriptions(String topic, long expireTimeInSeconds)
      */
     List<Message<byte[]>> peekMessages(String topic, String subName, int numMessages) throws PulsarAdminException;
 
+    /**
+     * Peek messages from a topic subscription.
+     *
+     * @param topic
+     *            topic name
+     * @param subName
+     *            Subscription name
+     * @param numMessages
+     *            Number of messages
+     * @param showServerMarker
+     *            Enables the display of internal server write markers
+     * @param transactionIsolationLevel
+     *            Sets the isolation level for peeking messages within transactions.
+     *            - 'READ_COMMITTED' allows peeking only committed transactional messages.
+     *            - 'READ_UNCOMMITTED' allows peeking all messages,
+     *                                 even transactional messages which have been aborted.
+     * @return
+     * @throws NotAuthorizedException
+     *             Don't have admin permission
+     * @throws NotFoundException
+     *             Topic or subscription does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    List<Message<byte[]>> peekMessages(String topic, String subName, int numMessages,
+                                       boolean showServerMarker, SubscriptionIsolationLevel transactionIsolationLevel)
+            throws PulsarAdminException;
```

Comment:
It's better to add a default implementation for https://github.com/apache/pulsar/pull/22762/files#diff-943b19d9674155ee923cf360b1a85ded4fcb2708edb9dc89be8cbf0ba3124deaR1657

So that users can know the default value for `showServerMarker` and `transactionIsolationLevel`